prow/sidecar/run.go

```diff
@@ -204,11 +191,7 @@ func (o Options) doUpload(spec *downwardapi.JobSpec, passed, aborted bool, metad
 	}
 
 	// TODO(fejta): move to initupload and Started.Repos, RepoVersion
-	if spec.Refs != nil {
-		finished.Revision = getRevisionFromRef(spec.Refs)
-	} else if len(spec.ExtraRefs) > 0 {
-		finished.Revision = getRevisionFromRef(&spec.ExtraRefs[0])
-	}
+	finished.Revision = downwardapi.GetRevisionFromSpec(spec)
```

Comment:
I would like us to drop Revision as it is not useful:
* job-version in finished.json needs to be set to what the job configures at runtime and puts in metadata.json
* repo-version should be in started.json, set from the refs
* we should have some other field that does a more complex recording of the refs (either repos or else a new refs field).

Revision just duplicates repo-version in the wrong location.

Reply:
I added that when we were thinking we should only keep a timestamp in started.json. Since I also changed places like testgrid and gubernator, I need to make some clean-up PRs to follow up.

Reply:
> * we should have some other field that does a more complex recording of the refs (either repos or else a new refs field).

Agreed, although I'm not sure that repos is sufficient. The complete ref data also includes ref names and any pull request numbers and SHAs. Including the complete serialized list of `Refs` structs in a `refs` field may be the best option.

Reply:
Does that mean testgrid need to know a prow struct? Note that we also need to handle other hand-crafted started.json :-)

Maybe we can dump the refs into metadata?